src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -603,109 +621,154 @@ protected boolean setupAndConsumeRecipeInputs(@NotNull Recipe recipe,
      *
-     * @param recipe          - The Recipe that will be consumed from the inputs and ran in the machine
-     * @param importInventory - The inventory that the recipe should be consumed from.
-     *                        Used mainly for Distinct bus implementation for multiblocks to specify
-     *                        a specific bus
-     * @return - true if the recipe is successful, false if the recipe is not successful
-     */
-    protected boolean setupAndConsumeRecipeInputs(@NotNull Recipe recipe,
-                                                  @NotNull IItemHandlerModifiable importInventory) {
+     * @param recipe          The Recipe that will be consumed from the inputs and ran in the machine
+     * @param importInventory The inventory that the recipe should be consumed from. Used mainly for Distinct bus
+     *                        implementation for multiblocks to specify a specific bus
+     * @return the recipe if the setup is successful, null if the setup is not successful
+     */
+    @MustBeInvokedByOverriders
+    protected @Nullable Recipe setupAndConsumeRecipeInputs(@NotNull Recipe recipe,
+                                                           @NotNull IItemHandlerModifiable importInventory) {
         return setupAndConsumeRecipeInputs(recipe, importInventory, this.getInputTank());
     }
 
     /**
-     * @param resultOverclock the overclock data to use. Format: {@code [EUt, duration]}.
+     * Overclock a recipe beyond a duration of 1 tick using parallelization.
+     *
+     * @param ocResult        the result of the overclock
+     * @param recipe          the recipe to overclock
+     * @param importInventory the input item inventory
+     * @param importFluids    the input fluid inventory
+     * @return the recipe if a valid recipe is produced, otherwise null
+     */
+    protected @Nullable Recipe subTickOC(@NotNull OCResult ocResult, @NotNull Recipe recipe,
+                                         @NotNull IItemHandlerModifiable importInventory,
+                                         @NotNull IMultipleTankHandler importFluids) {
+        RecipeMap<?> map = getRecipeMap();
+        if (map == null) {
+            return null;
+        }
+
+        Recipe r = new RecipeBuilder<>(recipe, map)
+                .EUt(ocResult.eut())
+                .build()
+                .getResult();
+
+        if (r == null) {
+            // should be impossible, but check anyway
+            return recipe;
+        }
+
+        RecipeBuilder<?> builder = findMultipliedParallelRecipe(map, r, importInventory, importFluids,
+                getOutputInventory(), getOutputTank(), ocResult.parallel(), ocResult.parallelEUt(),
+                getMetaTileEntity());
+
+        if (builder == null) {
+            return null;
+        }
+
+        if (builder.getParallel() == 0) {
+            return recipe;
+        }
+
+        ocResult.setEut(builder.getEUt());
+        r = builder.EUt(recipe.getEUt())
```

Comment:
I thought that in the parallel subtickOC case that the energy used was increased, at least according to the opening statement of this PR. If that is the case, then why do we use the pre-paralleled EUt value here?